<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>FliterWeight</class>
 <widget class="QWidget" name="FliterWeight">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>704</width>
    <height>498</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <widget class="QWidget" name="gridLayoutWidget">
   <property name="geometry">
    <rect>
     <x>60</x>
     <y>50</y>
     <width>201</width>
     <height>171</height>
    </rect>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <item row="2" column="0">
     <widget class="QLineEdit" name="kernel7">
      <property name="text">
       <string>1</string>
      </property>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QLineEdit" name="kernel1">
      <property name="text">
       <string>1</string>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QLineEdit" name="kernel2">
      <property name="text">
       <string>1</string>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QLineEdit" name="kernel4">
      <property name="text">
       <string>1</string>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QLineEdit" name="kernel5">
      <property name="text">
       <string>1</string>
      </property>
     </widget>
    </item>
    <item row="2" column="1">
     <widget class="QLineEdit" name="kernel8">
      <property name="text">
       <string>1</string>
      </property>
     </widget>
    </item>
    <item row="0" column="2">
     <widget class="QLineEdit" name="kernel3">
      <property name="text">
       <string>1</string>
      </property>
     </widget>
    </item>
    <item row="1" column="2">
     <widget class="QLineEdit" name="kernel6">
      <property name="text">
       <string>1</string>
      </property>
     </widget>
    </item>
    <item row="2" column="2">
     <widget class="QLineEdit" name="kernel9">
      <property name="text">
       <string>1</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="gridLayoutWidget_2">
   <property name="geometry">
    <rect>
     <x>90</x>
     <y>240</y>
     <width>131</width>
     <height>101</height>
    </rect>
   </property>
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0">
     <widget class="QLabel" name="label">
      <property name="text">
       <string>系数</string>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QLineEdit" name="coefficient">
      <property name="text">
       <string>1</string>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QLineEdit" name="common">
      <property name="text">
       <string>128</string>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QLabel" name="label_2">
      <property name="text">
       <string>常数或阈值</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="verticalLayoutWidget">
   <property name="geometry">
    <rect>
     <x>400</x>
     <y>60</y>
     <width>160</width>
     <height>268</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QPushButton" name="Inversion">
      <property name="text">
       <string>底片效果(反色)</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="meanBlur">
      <property name="text">
       <string>均值模糊</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="gaussian">
      <property name="text">
       <string>高斯模糊</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="medianFliter">
      <property name="text">
       <string>中值滤波</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="autoMedianFliter">
      <property name="text">
       <string>自适应中值滤波</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="lowPathFliter">
      <property name="text">
       <string>低通滤波</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="sharpen">
      <property name="text">
       <string>锐化</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="emboss">
      <property name="text">
       <string>浮雕</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QLabel" name="label_3">
   <property name="geometry">
    <rect>
     <x>320</x>
     <y>30</y>
     <width>72</width>
     <height>15</height>
    </rect>
   </property>
   <property name="text">
    <string>设置算子</string>
   </property>
  </widget>
  <widget class="QPushButton" name="ShowOrigin">
   <property name="geometry">
    <rect>
     <x>580</x>
     <y>170</y>
     <width>93</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>显示原图</string>
   </property>
  </widget>
  <widget class="QPushButton" name="apply">
   <property name="geometry">
    <rect>
     <x>580</x>
     <y>210</y>
     <width>93</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>应用</string>
   </property>
  </widget>
  <widget class="QGroupBox" name="groupBox">
   <property name="geometry">
    <rect>
     <x>310</x>
     <y>330</y>
     <width>311</width>
     <height>171</height>
    </rect>
   </property>
   <property name="title">
    <string>边缘检测</string>
   </property>
   <widget class="QWidget" name="horizontalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>40</y>
      <width>291</width>
      <height>81</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QCheckBox" name="Sobel">
         <property name="text">
          <string>Sobel算子</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="Prewitt">
         <property name="text">
          <string>Prewitt算子</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="Canny">
         <property name="text">
          <string>Canny算子</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QCheckBox" name="Horizontal">
       <property name="text">
        <string>水平</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="Vertical">
       <property name="text">
        <string>垂直</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QPushButton" name="preview">
   <property name="geometry">
    <rect>
     <x>580</x>
     <y>130</y>
     <width>93</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>预览效果</string>
   </property>
  </widget>
  <widget class="QWidget" name="horizontalLayoutWidget_2">
   <property name="geometry">
    <rect>
     <x>60</x>
     <y>370</y>
     <width>194</width>
     <height>80</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_2">
    <item>
     <widget class="QPushButton" name="saveOperator">
      <property name="text">
       <string>保存算子</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="readOperator">
      <property name="text">
       <string>读取算子</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="horizontalLayoutWidget_3">
   <property name="geometry">
    <rect>
     <x>430</x>
     <y>20</y>
     <width>291</width>
     <height>31</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_3">
    <item>
     <widget class="QPushButton" name="addNoise">
      <property name="text">
       <string>添加噪声</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QCheckBox" name="salt">
      <property name="text">
       <string>焦噪声</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QCheckBox" name="random">
      <property name="text">
       <string>盐噪声</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
